### ISSUE TITLE
Need help with color picker after WP 4.9 update

### ISSUE THREAD

**Issue by goodwinpress** [gambitph/Titan-Framework#403]
Hi folks
WordPress 4.9 crush color options.

upd problem **solved** if replace old version of wp-color-picker-alpha from TF to new version from

https://github.com/kallookoo/wp-color-picker-alpha-plugin

**Comment by mathieupreaud** [comment]
Same problem for me, all my Color Picker options crush after actualizing. In which folder of Titan Framework do you have to replace the file wp-color-picker-alpha? I don't see any PHP file with that name.

**Comment by andrastudio** [comment]
I noticed there are two js files

js/wp-color-picker-alpha.js
js/dev/wp-color-picker-alpha.js

Already replaced both files and nothing changed, not sure if we have to update the wp-color-picker-alpha.php file as well. I don't see any file named wp-color-picker-alpha.php in TF package

**Comment by goodwinpress** [comment]
HI andrastudio

there is a path: titan-framework/js/min/wp-color-picker-alpha-min.js

or

master/js/min/wp-color-picker-alpha-min.js

delete all code in this file and replace with a code from https://github.com/kallookoo/wp-color-picker-alpha-plugin/blob/master/wp-color-picker-alpha.js

and clean the cache in browser options.

In my case - before replacement:

![02 19 41 13](https://user-images.githubusercontent.com/8179689/32938469-4e10a0aa-cb8d-11e7-90eb-21295ee0b09c.png)

and after replacement:

![01 19 41 13](https://user-images.githubusercontent.com/8179689/32938509-6f9f67f6-cb8d-11e7-824e-2d52568906eb.png)

**Comment by andrastudio** [comment]
Thanks @goodwinpress, works like a charm!

**Comment by goodwinpress** [comment]
Glad to do my part.

**Comment by Khoapq** [comment]
I looks like TitanFramework is no longer maintaining by author?

**Comment by goodwinpress** [comment]
> I looks like TitanFramework is no longer maintaining by author?

Who knows?
TF is free, may be people from Gambith has no enough motivation to further development.
I have more than 3000 clients using TF in my themes and I'd like to see perspective of TF more clearly

**Comment by lockstockd** [comment]
@goodwinpress your fix has worked great thank you so much for this!

I'm now seeing something strange with the font type setting where it's now showing two color selectors:

![example-1](https://user-images.githubusercontent.com/34384591/33787595-e1c2d51e-dc65-11e7-8d0d-9e240c21060f.png)

My code:

`$headerTab->createOption( array(
		'name'                => __('Header Font', 'my-plugin'),
		'id'                  => 'header_font',
		'type'                => 'font',
		'classname'           => 'header-istext',
		'default'             => array( 'font-size' => '24px', 'color' => '#ffffff' ),
		'show_font_weight'    => false,
		'show_font_style'     => false,
		'show_line_height'    => false,
		'show_letter_spacing' => false,
		'show_text_transform' => false,
		'show_font_variant'   => false,
		'show_text_shadow'    => false,
		'show_preview'        => false
	) );`

Anyone else seeing this or is it just me?

**Comment by ahmadawais** [comment]
Same problem here 

![image](http://on.ahmda.ws/ogll/c)

**Comment by ahmadawais** [comment]
Guys, this PR with the following new branch fixes the issue 
https://github.com/gambitph/Titan-Framework/pull/405

**Comment by majick777** [comment]
@lockstockd  have the same with you for the font color selectors (though it probably should be a separate issue)... here's what I did to fix it...

in /lib/class-option-font.php, wrap both color AND text shadow color inputs in a styled div:
```
<div style="float:right; margin-left:10px; vertical-align:middle;">
    <input class='tf-font-sel-{shadow-}color' ......
</div>
```

Then I added this just after the "Initialize color picker code" javascript:
```
$('.tf-font .tf-font-sel-shadow-color').removeClass('tf-font-sel-shadow-color')
.parent().addClass('tf-font-sel-shadow-color');
```
which fixes the show/hide code for the text shadow color input. :-)

I probably should do a pull request - seems a bit hacky but works.

**Comment by bfintal** [comment]
Fixed by #404 #405 #411

**Comment by lockstockd** [comment]
Thank you all so much for fixing this!
